src/Collection.js

```diff
@@ -1,542 +1,611 @@
-// Create a local reference to a common array method we'll want to use later.
-const slice = Array.prototype.slice;
+import { slice, splice, wrapError } from './utils';
+import extend from './extend';
+import { Model } from './Model';
+import { sync } from './sync';
+import _ from 'lodash';
+import { mixinMethods } from './mixin-methods';
+import { Events } from './events';
 
-// Backbone.Collection
-// -------------------
+// Collection
+// ----------
 
-// If models tend to represent a single row of data, a Backbone Collection is
-// more analogous to a table full of data ... or a small slice or page of that
-// table, or a collection of rows that belong together for a particular reason
+// Defining an @@iterator method implements JavaScript's Iterable protocol.
+// In modern ES2015 browsers, this value is found at Symbol.iterator.
+const $$iterator = typeof Symbol === 'function' && Symbol.iterator;
+
+// If models tend to represent a single row of data, a Collection is more
+// analogous to a table full of data ... or a small slice or page of that table,
+// or a collection of rows that belong together for a particular reason
 // -- all of the messages in this particular folder, all of the documents
-// belonging to this particular author, and so on. Collections maintain
-// indexes of their models, both in order, and for lookup by `id`.
+// belonging to this particular author, and so on. Collections maintain indexes
+// of their models, both in order, and for lookup by `id`.
+
+// Default options for `Collection#set`.
+const setOptions = { add: true, remove: true, merge: true };
+const addOptions = { add: true, remove: false };
 
 // Create a new **Collection**, perhaps to contain a specific type of `model`.
 // If a `comparator` is specified, the Collection will maintain
 // its models in sort order, as they're added and removed.
-const Collection = (Backbone.Collection = function (models, options) {
-  options || (options = {});
-  this.preinitialize.apply(this, arguments);
-  if (options.model) this.model = options.model;
-  if (options.comparator !== void 0) this.comparator = options.comparator;
-  this._reset();
-  this.initialize.apply(this, arguments);
-  if (models) this.reset(models, _.extend({ silent: true }, options));
-});
+export const Collection = /*#__PURE__*/ (() => {
```

Comment:
Is `/*#__PURE__*/` a special directive (e.g. for linters)?

Reply:
it's for tools like `terser` and `babel` to understand what they can remove if unused. otherwise, they assume even simple things like property assignment are side-effectful and refuse to remove them.